packages/core/admin/admin/src/content-manager/hooks/useLazyComponents/index.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import { useCustomFields } from '@strapi/helper-plugin';
+
+/**
+ * @description
+ * A hook to lazy load custom field components
+ * @param {Array.<string>} componentUids - The uids to look up components
+ * @returns object
+ */
+const useLazyComponents = (componentUids) => {
+  const [lazyComponentStore, setLazyComponentStore] = useState({});
+  const [loading, setLoading] = useState(true);
+  const customFieldsRegistry = useCustomFields();
+
+  useEffect(() => {
+    const lazyLoadComponents = async (uids, components) => {
```

Comment:
ℹ️ Two independent arrays that then become one object feels a bit strange but I didn't see another way to make it work with `Promise.all`. I think it is reliable since Promise.all returns them in the same order, but open to other suggestions.